ast/src/asm_analysis/mod.rs

```diff
@@ -570,11 +570,20 @@ impl<T> From<Return<T>> for FunctionStatement<T> {
 #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub struct AssignmentStatement<T> {
     pub start: usize,
-    pub lhs: Vec<String>,
-    pub using_reg: Option<String>,
+    pub lhs_with_reg: Vec<(String, OutputRegister)>,
```

Comment:
Does it make sense to group it like that? It's probably convenient, but it could be grouped with rhs in the same way...

Reply:
Not really, the RHS is of type `Box<Expression<T>>`, so it's just a single expression (which might evaluate to a list of values...)